### PR TITLE
feat(ci): Add automated changelog workflows to repo

### PR DESCRIPTION
**Pull request by samiralavi**
This PR adds the `CHANGELOG.md` file to the repository and the GitHub workflow that updates it on every PR closure. The Github action is the same one used in FastAPI: https://github.com/tiangolo/latest-changes. The way it works is that only the PRs labelled with the following ones are added to the corresponding section:

* `breaking`: `Breaking Changes`
* `security`: `Security Fixes`
* `feature`: `Features`
* `bug`: `Fixes`
* `refactor`: `Refactors`
* `upgrade`: `Upgrades`
* `docs`: `Docs`
* `lang-all`: `Translations`
* `internal`: `Internal`

I have also updated the release workflow to automatically update the release version in the `CHANGELOG.md` file.

### CODE REVIEW

**Comment by samiralavi** [comment]
@safeamiiir should I create new labels on the repo based on the default ones mentioned here?

File: .github/workflows/changelog.yml
Comment (safeamiiir) [inline .github/workflows/changelog.yml:1]: Can we run this workflow after each release instead of each PR into main?

**Review by safeamiiir — CHANGES_REQUESTED** [review]

File: .github/workflows/changelog.yml
Comment (samiralavi) [reply .github/workflows/changelog.yml:1]: You are right, running it after every commit is overkill. I am going to use a different github action to only add items when we release.

**Review by str-anger — APPROVED** [review]

**Review by jatinriverlane — COMMENTED** [review]
Thanks @samiralavi

**Review by jatinriverlane — APPROVED** [review]

**Review by str-anger — APPROVED** [review]

**Comment by samiralavi** [comment]
@safeamiiir I have made the changes to generate and commit the changelog only on release. It creates there categories for breaking changes, new features and fixes so anyone can easily find the changes.

**Review by safeamiiir — APPROVED** [review]
Nice and simple!
Thanks

**Pull request merged**